### PR TITLE
TheBotArrives

### PR DESCRIPTION
**Pull request by MateiPribagu**
=== SINGLEPLAYER BOT ===
Bot plays cards before taking actions
Bot plays cards in descending order of their cost, prioritizing Airdrop
Bot knows to skip to taking actions when all Beacon summoning places are occupied, which is why it loves Airdrop
Bot moves and attacks randomly
=== BONUS ===
Fixed a bug where Anubite would act as if he had Guard, because the second parameter of HasKeword was not multiplied by 1 when checking Guard
Players now get a job done message and sound when they don't have anything left to do
Made a chat log system to end the scourge of god power spam

### CODE REVIEW

File: shared.c [hunk 1]
```diff
@@ -496,6 +498,27 @@ void DeploySober(string p="", string v=""){
...
+	for(i=9;>=0){
+		trChatSendToPlayer(0, p, trStringQuestVarGet("chat" + p + "Log" + i));
+	}
+}
+
+void ChatLog(int p = 1, string message = ""){
+	if (trCurrentPlayer() == p) {
+		trChatHistoryClear();
+	}
+	for(i=9;>0){
```
[the remark this answers is not in the document]
Comment (MateiPribagu) [reply shared.c]: done

File: gameplay.c [hunk 2]
```diff
@@ -130,14 +165,26 @@ void findTargets(int index = 0, string db = "") {
...
 			}
 		}
 	}
+	if(anyTargets){
+		// Bot attacks
+		if(Bot && trQuestVarGet("p2jobDoneHand") == 1){
+			BotClickRight();
+		}
+	}
 }
```
Comment (jkalexho220) [inline gameplay.c:189]: Everything here can be achieved without modifying findTargets().

1. The bot calls the findTargets() function
2. To check if there are anyTargets, just check the size of the database created by findTargets. yGetDatabaseCount() > 0
3. Then the bot picks a target by using a database search and its own selection criteria.

File: shared.c [hunk 3]
```diff
@@ -496,6 +498,24 @@ void DeploySober(string p="", string v=""){
 	trUnitTeleport(trVectorQuestVarGetX(v),trVectorQuestVarGetY(v),trVectorQuestVarGetZ(v));
 }
 
+void ChatLogShow(int p = 1){
+	if (trCurrentPlayer() == p) {
+		trChatHistoryClear();
+	}
+	int x = peekModularCounterNext("chat" + p + "Log");
+	for(i=x;<11){
```
Comment (jkalexho220) [inline shared.c:506]: just use a single for loop and replace i with modularCounterNext() lol. 
`for(i=11; >0) {
trChatSendToPlayer(0, p, trStringQuestVarGet("chat" + p + "log" + modularCounterNext("chat" + p + "log"));
}`

Comment (jkalexho220) [reply shared.c:506]: Doesn't actually matter since the runtime is the same but w/e

**Pull request closed without merging**